ethcontract/src/transaction/confirm.rs

```diff
@@ -24,23 +25,37 @@ pub struct ConfirmParams {
     /// values indicate that extra blocks should be waited for on top of the
     /// block where the transaction was mined.
     pub confirmations: usize,
-    /// The polling interval. This is used as the interval between consecutive
-    /// `eth_getFilterChanges` calls to get filter updates, or the interval to
-    /// wait between confirmation checks in case filters are not supported by
-    /// the node (for example when using Infura over HTTP(S)).
-    pub poll_interval: Duration,
+    /// Minimal delay between consecutive `eth_blockNumber` calls.
+    /// We wait for transaction confirmation by polling node for latest
+    /// block number. We use exponential backoff to control how often
+    /// we poll the node.
+    pub poll_interval_min: Duration,
+    /// Maximal delay between consecutive `eth_blockNumber` calls.
+    pub poll_interval_max: Duration,
+    /// Factor, by which the delay between consecutive `eth_blockNumber`
+    /// calls is multiplied after each call.
+    pub poll_interval_factor: f32,
     /// The maximum number of blocks to wait for a transaction to get confirmed.
     pub block_timeout: Option<usize>,
 }
 
-/// The default poll interval to use for confirming transactions.
-///
-/// Note that this is currently 7 seconds as this is what was chosen in `web3`
-/// crate.
+/// Default minimal delay between polling the node for transaction confirmation.
+#[cfg(not(test))]
+const DEFAULT_POLL_INTERVAL_MIN: Duration = Duration::from_millis(250);
+#[cfg(test)]
+const DEFAULT_POLL_INTERVAL_MIN: Duration = Duration::from_millis(0);
+
+/// Default maximal delay between polling the node for transaction confirmation.
 #[cfg(not(test))]
-pub const DEFAULT_POLL_INTERVAL: Duration = Duration::from_secs(7);
+const DEFAULT_POLL_INTERVAL_MAX: Duration = Duration::from_millis(7000);
 #[cfg(test)]
-pub const DEFAULT_POLL_INTERVAL: Duration = Duration::from_secs(0);
+const DEFAULT_POLL_INTERVAL_MAX: Duration = Duration::from_millis(0);
+
+/// Default factor for increasing delays between node polls.
+#[cfg(not(test))]
+const DEFAULT_POLL_INTERVAL_FACTOR: f32 = 1.7;
```

Comment:
Just curious about the choice of `f32` over `f64`, not that it really matters :stuck_out_tongue:.

Reply:
Idk, we don't need any precision here, so I choose the smallest thing possible. Sort of like the default int is `i32`.

Reply:
Cool just curious if there was a specific reason for it.